### PR TITLE
Add youtube-music-desktop-app-controller extension

### PR DESCRIPTION
**Pull request by muditmahajan21**
## Description

Added an extension to control unofficial Youtube Music Desktop client available at https://ytmdesktop.app/

Adds various commands such as Play, Pause, Next Track,Previous Track etc. 
First time auth setup withe the companion server of YTMDesktop

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/youtube-music-desktop-app-controller/package.json [hunk 1]
```diff
@@ -0,0 +1,145 @@
+{
+  "$schema": "https://www.raycast.com/schemas/extension.json",
+  "name": "youtube-music-desktop-app-controller",
+  "title": "Youtube Music Desktop App Controller",
```
Comment (pernielsentikaer) [inline extensions/youtube-music-desktop-app-controller/package.json:4]:
```suggestion
  "title": "Youtube Music Desktop",
```
What about this name/title

File: extensions/youtube-music-desktop-app-controller/package.json [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+      "name": "playPause",
+      "title": "Play/Pause",
+      "subtitle": "Play/Pause for Youtube Music Desktop App",
+      "description": "Play or pause the current track in Youtube Music Desktop App",
+      "mode": "no-view"
+    },
+    {
+      "name": "play",
+      "title": "Play",
+      "subtitle": "Play for Youtube Music Desktop App",
```
Comment (pernielsentikaer) [inline extensions/youtube-music-desktop-app-controller/package.json:27]: This should show the name of the extension, it's not for describing - the same goes for all commands 🙂

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you:

- I don't currently have the application installed, can we have something like how [Eagle](https://www.raycast.com/Yukai/eagle) handles it?

- All `throw`'s should be handled gracefully

- I think we should add `"disabledByDefault": true,` for a good amount of the commands like we do in the [Spotify extension](https://www.raycast.com/mattisssa/spotify-player)

- If you using a no-view command we might want to use `launchCommand` to get started with the auth

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

**Comment by muditmahajan21** [comment]
Ah thanks for the review!

Let me take a look at the things over the weekend :)

File: extensions/youtube-music-desktop-app-controller/package.json (hunk 2 above)
Comment (muditmahajan21) [reply extensions/youtube-music-desktop-app-controller/package.json]: Updated in https://github.com/raycast/extensions/pull/14379/commits/91fad14bf6a729551bf62f7b6826997bdbf8c9ee

File: extensions/youtube-music-desktop-app-controller/package.json (hunk 1 above)
Comment (muditmahajan21) [reply extensions/youtube-music-desktop-app-controller/package.json]: Updated in https://github.com/raycast/extensions/pull/14379/commits/91fad14bf6a729551bf62f7b6826997bdbf8c9ee

**Comment by muditmahajan21** [comment]
> I'm not really sure I like the idea of having an extension for an application that requires forcing macOS to install, especially since it's not signed and might contain malware. How certain are we about this?

I see, well I cannot say anything about the YTM Desktop application as I am just a user of that and Raycast and wanted a way to control that from raycast so created the extension. 

If there are security concerns for this, I am happy to close the request and just use the dev version in my local setup.

**Comment by pernielsentikaer** [comment]
- When pausing I see two HUDs at once 

Raycast18092024-U9cn4Ts3 png@2x

- Maybe call `YTMdesktop Setup` something like `Authenticate` instead so it's easier for the user to figure out

**Pull request closed without merging**